### PR TITLE
feat(carousel): L3-4045 carousel overflow and zoom/pan behavior

### PR DESCRIPTION
**Pull request by sgrund14**
**Jira ticket**

[L3-4045](https://phillipsauctions.atlassian.net/browse/L3-4045)

**Screenshots**

**Dot Overflow**
https://github.com/user-attachments/assets/b0f74550-9de5-4cc5-b05c-733c3fcf4a10

**Carousel when item is zoomed and at edge**

https://github.com/user-attachments/assets/d0be4a68-359d-445b-bbae-b4668af9d0b6

**Summary**

Add carousel dots overflow behavior

https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?m=auto&node-id=15894-140133&focus-id=15894-140065

And the ability to pan to the next carousel item when zoomed in and at the edge of an item

**Change List (describe the changes made to the files)**

- Use intersection observer in CarouselDots & CarouselDot to track dot visibility and shrink/unshrink
- Add tracking to the PinchZoom component to detect edges
- Add stories which show how to use to enable pan-when-zoomed-to-edge behavior

**Acceptance Test (how to verify the PR)**

- Add step by step instructions on how to verify the change

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-4045]: https://phillipsauctions.atlassian.net/browse/L3-4045?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Carousel/CarouselDot.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import classNames from 'classnames';
+import { useInView } from 'react-intersection-observer';
+
+type Props = {
+  isSelected: boolean;
+  onClick: () => void;
+  baseClassName: string;
+  scrollableContainerRef: React.RefObject<HTMLDivElement>;
+  onInViewChange: (inView: boolean) => void;
+  isShrinked: boolean;
```
Comment (scottdickerson) [inline src/components/Carousel/CarouselDot.tsx:10]: might be better to do a `variant` `sm` or `lg` here

File: src/components/Carousel/CarouselDot.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import classNames from 'classnames';
+import { useInView } from 'react-intersection-observer';
+
+type Props = {
```
Comment (davidicus) [inline src/components/Carousel/CarouselDot.tsx:4]: why the type and not an interface?

Comment (davidicus) [reply src/components/Carousel/CarouselDot.tsx:4]: should be an interface with comments so we get the documentation in storybook.

File: src/components/Carousel/CarouselDot.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <button
+      ref={ref}
+      role="button"
+      onClick={onClick}
+      className={classNames(`${baseClassName}-pagination-dot-container`)}
+    >
+      <span
+        className={classNames(`${baseClassName}-pagination-dot`, {
+          [`${baseClassName}-pagination-dot-selected`]: isSelected,
```
Comment (davidicus) [inline src/components/Carousel/CarouselDot.tsx:38]: BEM has a `--` for modifier.
```suggestion
          [`${baseClassName}-pagination-dot--selected`]: isSelected,
          [`${baseClassName}-pagination-dot--${variant}`]: variant,
```

File: src/components/Carousel/CarouselDot.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      onInViewChange(inView);
+    },
+  });
+
+  return (
+    <button
+      ref={ref}
+      role="button"
+      onClick={onClick}
+      className={classNames(`${baseClassName}-pagination-dot-container`)}
```
Comment (davidicus) [inline src/components/Carousel/CarouselDot.tsx:34]: BEM has a `__` for elements.

```suggestion
      className={classNames(`${baseClassName}-pagination-dot__container`)}
```

File: src/components/Carousel/_carousel.scss [hunk 5]
```diff
@@ -60,24 +60,45 @@
...
+        display: none; /* Safari and Chrome */
+      }
     }
 
     &-dot-container {
       align-items: center;
       display: flex;
       height: 8px;
       justify-content: center;
+      min-width: 8px;
```
Comment (davidicus) [inline src/components/Carousel/_carousel.scss:83]: Do we need this if we are explicitly setting the width?

**Review by davidicus — COMMENTED** [review]
awesome work. Some very minor comments. I see there are pre-existing BEM issues which, for consistency, would like to correct but not necessary right now.

File: src/components/Carousel/CarouselDot.tsx (hunk 2 above)
Comment (sgrund14) [reply src/components/Carousel/CarouselDot.tsx:4]: this is internal to to `CarouselDots`, which is the exposed to consumers, so there's not actually a storybook for it

i can update to interface tho

File: src/components/Carousel/_carousel.scss (hunk 5 above)
Comment (sgrund14) [reply src/components/Carousel/_carousel.scss:83]: there's this weird thing where the dots gets squashed without it 🤷‍♂️

**Pull request merged**